source/slang/ir-legalize-types.cpp

```diff
@@ -1360,10 +1384,9 @@ static LegalVal declareSimpleVar(
 
     case kIROp_Var:
         {
+            builder->setInsertBefore(context->insertBeforeLocalVar);
```

Comment:
Instead of having `emitVar` temporary inserting the newly created var inst into an undefined place then remove it, just set up builder properly to have `emitVar` insert it directly into the intended place.

Reply:
This seems like a good fix, yeah.